### DISCUSSION TITLE
v0.8.0

### DISCUSSION THREAD

**Discussion by Blinue** [Blinue/Magpie#252] [Announcements]
v0.8 要求安装 [.NET 6 运行时](https://dotnet.microsoft.com/download)。

**新功能**

1. 用户界面迁移到 .NET 6
2. 支持裁剪 UWP 窗口的标题栏。一些 UWP 窗口的标题栏是用户界面的一部分，因此提供了关闭这个功能的选项
3. 支持多显示器
   - 支持 DPI 缩放不同的多显示器
   - 支持全屏窗口跨越多个显示器 #219 
   - 全屏时光标可以在显示器间自由移动，此时操作其他显示器上的窗口将不会退出全屏状态 #65 
4. 添加新的捕获模式 Desktop Duplication，要求操作系统为 win10 v2004 或更新 👉[现有捕获模式的对比](https://github.com/Blinue/Magpie/wiki/捕获模式对比)
6. 添加模拟独占全屏的功能，可以起到免打扰的效果 #245 

**效果变更**

1. Mitchell 重命名为 Bicubic，参数不再兼容
2. 添加 NIS 和 NVSharpen，它们的使用示例见 https://github.com/Blinue/Magpie/issues/202#issuecomment-981352269
3. 添加一些 CRT 着色器，包括 CRT-Easymode, CRT-Geom, CRT-Hyllian, CRT-Lottes 和 GTU #160 
4. 添加 Anime4K_Thin_HQ 和 Anime4K_Upscale_GAN_x2_S
5. 添加 LumaSharpen 和 FineSharp #242 
6. 添加 FSR_EASU_DX10，它是支持 DirectX 功能级别 10 的 FSR_EASU，速度比原始版本稍慢

感谢 @hooke007 维护一个 Magpie 的所有效果的示例配置 https://gist.github.com/hooke007/818ecc88f18e229bca743b7ae48947ad

**其他更改**

1. 优化了 Graphics Capture 捕获模式的兼容性
2. 优化了帧率在白色背景下的可见性 #111 
3. 现在可以安全的在其他程序/脚本中启动 Magpie 而不必设置工作路径 #236 

**系统需求**

1. Windows 10/11
2. DirectX 功能级别 10
3. [.NET 6 运行时](https://dotnet.microsoft.com/download)
4. [MSVC 2015-2022 运行时](https://docs.microsoft.com/cpp/windows/latest-supported-vc-redist)

**注意事项**

1. v0.8 是最后一个支持 DirectX 功能级别 10 的版本
2. Magpie 的实现中 NIS 比 FSR 慢的多，但是 NVSharpen 相当快

<hr /><em>This discussion was created from the release <a href='https://github.com/Blinue/Magpie/releases/tag/v0.8.0'>v0.8.0</a>.</em>

**Comment by Blinue** [comment]
https://github.com/Z-Dante/Magpie 提供更多缩放配置

**Comment by amateur0000000** [comment]
"To run this application, you must install missing frameworks for .NET"

Even though I already installed all requirements? Please fix.

**Reply by Blinue** [reply]
https://dotnet.microsoft.com/en-us/download

You need to install .NET 6 Runtime.

**Reply by amateur0000000** [reply]
I have but didn't work. I had to Install .NET 6 SDK Version too for it to work. https://dotnet.microsoft.com/en-us/download/dotnet/6.0
You only have a link for the Runtime not SDK in description, so I missed it.
Thanks again for great app.

**Reply by hooke007** [reply]
Cannot reporduce.
Tested in a VM, after the installation of desktop net6 runtime, it would work.
![Snipaste_2021-12-28_21-38-55](https://user-images.githubusercontent.com/41094733/147572120-b3cd3b13-0556-4c1b-93e2-03dc887c7bbe.jpg)

**Reply by Blinue** [reply]
The requirement is ".NET 6.0 Desktop Runtime".  I've updated the links, thanks for the report!

**Comment by neavo** [comment]
Anime4K_Upscale_M 和 Anime4K_Restore_M 组合使用时似乎会报错

Profile:
`
  {
    "name": "2D - 一次放大",
    "effects": [
      {
        "effect": "Jinc",
        "scale": [
          -0.5,
          -0.5
        ]
      },
      {
        "effect": "Anime4K_Restore_M"
      },
      {
        "effect": "Anime4K_Upscale_M"
      },
      {
        "effect": "NVSharpen"
      }
    ]
  },
`

LOG：
`
2021-12-29 17:58:34.303|info|Renderer.cpp:Renderer::_CreateSwapChain|Hardware Composition 支持：是
2021-12-29 17:58:34.303|info|Renderer.cpp:Renderer::_CreateSwapChain|Multiplane Overlay 支持：是
2021-12-29 17:58:34.505|info|GraphicsCaptureFrameSource.cpp:GraphicsCaptureFrameSource::Initialize|GraphicsCaptureFrameSource 初始化完成
2021-12-29 17:58:34.505|info|App.cpp:App::Run|源窗口客户区尺寸：640x480
2021-12-29 17:58:34.505|info|EffectDrawer.cpp:EffectDrawer::Initialize|编译 effects\Jinc.hlsl 用时 0.262 毫秒
2021-12-29 17:58:34.506|info|EffectDrawer.cpp:EffectDrawer::Initialize|编译 effects\Anime4K_Restore_M.hlsl 用时 0.388 毫秒
2021-12-29 17:58:34.506|error|Utils.cpp:Utils::ReadTextFile|打开文件effects\Anime4K_Upscale_M.hlsl失败
2021-12-29 17:58:34.506|error|EffectCompiler.cpp:EffectCompiler::Compile|读取源文件失败
2021-12-29 17:58:34.506|error|EffectDrawer.cpp:EffectDrawer::Initialize|编译 effects\Anime4K_Upscale_M.hlsl 失败
2021-12-29 17:58:34.506|error|Renderer.cpp:Renderer::InitializeEffectsAndCursor|_ResolveEffectsJson 失败
2021-12-29 17:58:34.506|critical|App.cpp:App::Run|初始化效果失败，即将退出
2021-12-29 17:58:34.525|info|App.cpp:App::_Run|开始接收窗口消息
2021-12-29 17:58:34.528|info|CursorDrawer.cpp:CursorDrawer::~CursorDrawer|CursorDrawer 已析构
2021-12-29 17:58:34.544|info|App.cpp:App::_OnQuit|主窗口已销毁
2021-12-29 17:58:34.544|info|DllMain.cpp:Run|App.Run 失败
`

**Reply by Blinue** [reply]
Upscale 只移植了 S 和 L

**Reply by neavo** [reply]
> Upscale 只移植了 S 和 L

果然 😂

**Comment by Coliostro** [comment]
**Happy New Year, friend! And thanks for the new release from Russia!**
I tested it in all upscalers in all screen capture modes and compiled it into an OpenDocument (.odt) table. The impressions from the release are interesting, the FSR on my video card without DX11 still did not work. However, the results of the other upscalers also do not leave much to be desired... FPS on Original Full HD mode is higher, and the picture is nicer. Perhaps this is due to weak TMU blocks in the video card. But anyway, thank you very much for your diligence and enthusiasm. In the new year, I wish you good health, new interesting developments, and of course, that your work is rewarded not only with grateful comments, but also with USD donations. Make the details of your electronic card (Visa, Maestro) in a prominent place so that grateful users can thank you! 
[Magpie 8.0 Performance Test on old GPU (AMD Radeon HD 3850).ods](https://github.com/Blinue/Magpie/files/7796165/Magpie.8.0.Performance.Test.on.old.GPU.AMD.Radeon.HD.3850.ods)
[Magpie.log](https://github.com/Blinue/Magpie/files/7796166/Magpie.log)
[Runtime.log](https://github.com/Blinue/Magpie/files/7796167/Runtime.log)

**Reply by Blinue** [reply]
Thanks for the feedback and tests! Happy New Year

1. Magpie's built-in FSR does not support DX10, add this to ScaleModels.json then it should work:
``` json
{
    "name": "FSR-DX10",
    "effects": [
        {
            "effect": "FSR_EASU_DX10",
            "scale": [ -1, -1 ]
        },
        {
            "effect": "FSR_RCAS",
            "sharpness": 0.87
        }
    ]
}
```
2. There are currently no donation channels. If you've found this software helpful, please recommend it to others!

**Reply by Coliostro** [reply]
Ok!) I'll try, thanks!

**Comment by Ramatakka48** [comment]
please tell me, how to add nis and other method fx.. im really dont understand... i see lots FX like FXAA.. MSAA.. but i dont know how to use that.. please

**Reply by hooke007** [reply]
https://gist.github.com/hooke007/818ecc88f18e229bca743b7ae48947ad

**Reply by Ramatakka48** [reply]
good.. thanks a lot

**Reply by Ramatakka48** [reply]
sorry one more think.. its possible to combine FX... like.. lets say FSR but with SMAA..?

**Reply by hooke007** [reply]
yes u can check the examples of using multi-effects (in the end of conf file).

**Reply by hooke007** [reply]
```json
  {"name":"fsrfull-smaalow","effects":[
    {"effect":"FSR_EASU","scale":[-1,-1]},
    {"effect":"FSR_RCAS","sharpness":0.87},
    {"effect":"SMAA_Low"},
  ]},
```

**Reply by Ramatakka48** [reply]
ok thanks.. still learning but i think can do this

**Comment by NeilTohno** [comment]
贴一下自己弄的：
为视觉小说弄的轻锐化配置文件，新年快乐，(ﾉﾟ▽ﾟ)ﾉ
https://github.com/NeilTohno/Magpie/releases/

**Reply by Ramatakka48** [reply]
please.. why i cant use dekstop duplication method... its say minimum windows version is 2004.. but mine is 20h1 and still appears error message

**Reply by Blinue** [reply]
Please post the logs.

**Reply by mirh** [reply]
I'm having a hard time trying to understand what specific part of a W8 api would require a windows version from 8 years later?
